src/transformStateWithClones.js

```diff
@@ -6,8 +6,28 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let transormItem = { ...state };
+
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
```

Comment:
You can use `swith` here
```suggestion
  for (const i of actions) {
    switch (i.type) {
    case 'addProperties':
```

Reply:
And rename `i`